src/components/category/CategoryProvider.js

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, useState } from "react";
+
+export const CategoryContext = createContext()
+
+export const CategoryProvider = (props) => {
+    const [categories, setCategories] = useState([])
+
+    const getAllCategories = () => {
+        return fetch("http://localhost:8088/categories")
+        .then(res => res.json())
+        .then(setCategories)
+    }
+
+    const addCategory = catObj => {
+        return fetch("http://localhost:8088/categories", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(catObj)
+        })
+        .then(getAllCategories)
```

Comment:
Might be a bit safer/clearer to write `() => getAllCategories()`.

As written line 22 executes as `.then((res) => getAllCategories(res))`

To contrast, line 11 is perfect because it executes as `.then((jsonFromRes) => setCategories(jsonFromRes))` which is exactly what we want.

Reply:
Fixed in coming PR